platform/x11/cog-platform-x11.c

```diff
@@ -887,3 +904,58 @@ cog_platform_plugin_get_view_backend (CogPlatform   *platform,
 
     return wk_view_backend;
 }
+
+static void *
+check_supported(void *data G_GNUC_UNUSED)
+{
+    /*
+     * TODO: This could do more than only trying to connect to the X server.
+     */
+    Display *d = XOpenDisplay(NULL);
+    if (d) {
+        XCloseDisplay(d);
+        return GINT_TO_POINTER(TRUE);
+    } else {
+        return GINT_TO_POINTER(FALSE);
+    }
```

Comment:
This approach seems just fine. If the X protocol can work, then it's probably usable for presenting. Same would be done for the FDO module and the Wayland protocol that's used there, but a higher priority. This would avoid preferring the X11 module being picked to connect to Xwayland over the FDO module connecting to the Wayland display.

Reply:
OTOH, it would be good to at least check whether the X server can do EGL + OpenGLES, though it may not be reasonably simple to do for the `is_supported` check.

Regarding priorities, if you set `G_MESSAGES_DEBUG=Cog` in the environment before running Cog it will print the list of modules found and their priorities. Currently it shows:

```
Cog-INFO: 22:07:59.456: com.igalia.Cog.Platform:
Cog-INFO: 22:07:59.456:   fdo - 500/CogFdoPlatform
Cog-INFO: 22:07:59.456:   gtk4 - 400/CogGtk4Platform
Cog-INFO: 22:07:59.456:   x11 - 300/CogX11Platform
Cog-INFO: 22:07:59.456:   drm - 200/CogDrmPlatform
Cog-INFO: 22:07:59.456:   headless - 100/CogHeadlessPlatform
```

As you can see, the `fdo` module already has more priority than `x11`, which means native Wayland is already preferred over XWayland 😉